### ISSUE TITLE
Potential memory leak of DiagnosticListener - .NET Framework app on Azure App Service

### ISSUE THREAD

**Issue by jirikanda** [microsoft/ApplicationInsights-dotnet#2033]
I am creating a separate issue from #1867 to better target issue to the .NET Framework.

**Environment**

* SDK version: 2.14.0
* .NET runtime version (.NET Framework 4.7.2)
* Hosting Info (IIS/Azure WebApps/etc): Azure App Service
* Platform and OS version: Windows

**What is the expected behavior?**
GC able to reclaim memory.

**What is the actual behavior?**
Number of DiagnosticSubscriptions grows in time.

The are two images below:
* Memory consumption of our applications in time (more than two weeks).
* Memory dump analysis (debug) in Visual Studio 2019 of the application.

![memoryconsumption](https://user-images.githubusercontent.com/5111719/92364288-26639100-f0f3-11ea-9786-9bead93ec0a5.png)
![memorydebug](https://user-images.githubusercontent.com/5111719/92364292-26fc2780-f0f3-11ea-9007-bb01fbcdaae3.png)

**Comment by jirikanda** [comment]
Yes, good tactic.
No reaction. Autoclose. Solved.

Sad world.

**Comment by TimothyMothra** [comment]
Hello @jirikanda, I'm sorry this issue was overlooked when it was opened.

Do you have a repro for this issue? This would help us investigate and collect more data.

If not, you can create an Azure Support Request and one of our engineers can help collect a dump from your environment.

**Comment by cijothomas** [comment]
From a quick look, it looks like a code issue as it shows many instances of TelemetryClient itself. Please share the code snippet used to enable application insights, and we can review if it is correct.

**Comment by jirikanda** [comment]
To be honest, we have mitigated the issue.

But be sure there is no code which is creating a new TelemetryClient instance. It is hidden somewhere in the Application Insights packages.

But what we did was creating **a lot of** Azure Blob Storage Client (BlobContainerClient) instances. When we change this behavior to have BlobContainerClient singleton, the issue was mitigated.
There were some issues here on github trying to explain this. I was trying to explain this bad behavior by impacts - reusing BlobContainerClient caused to double memory consumption by TelemetryClient & co. in a week or two.

I am not going to provide any snippet after a year. I also did not checked if the issue with BlobContainerClient was mitigated in the library. Hope, you are not solving the issue as "not out team issue" on both sides.
